pkg/util/cmd.go

```diff
@@ -30,19 +32,37 @@ import (
 
 var ZapLogger *zap.Logger
 
+func BuildLogger(level int8, dev bool, w io.Writer) logr.Logger {
+	if dev {
+		ZapLogger = corezap.NewRaw(func(options *corezap.Options) {
+			level := zap.NewAtomicLevelAt(zapcore.Level(-level))
+			options.Level = &level
+			options.Development = dev
+		})
+	} else {
+		// we need to create ZapLogger manually because we don't want to use Sampler, that does not support arbitrary log levels
+		encoder := zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig())
+		sink := zapcore.AddSync(os.Stderr)
+		ZapLogger = zap.New(
```

Comment:
Add big warning somewhere we're setting this global variable here.

Reply:
I really don't like it, do you have any ideas on how to get rid of it?

Reply:
done